pkg/universe.dagger.io/x/cpuguy83@gmail.com/apt/install.cue

```diff
@@ -0,0 +1,115 @@
+	"dagger.io/dagger/core"
+	"universe.dagger.io/docker"
+)
+
+_#aptMounts: {
+	cache_prefix:     string
+	_conf_keep_cache: core.#WriteFile & {
+		input:    dagger.#Scratch
+		contents: "Binary::apt::APT::Keep-Downloaded-Packages \"true\";"
+		path:     "/keep-cache"
+	}
+
+	updated_path: "/var/cache/apt/dagger/updated"
+
+	// Use this to track if `apt-get update` needs to be re-run
+	// This works by checking the last update time of the file specified blow
+	// Since this is being written to a directory where ther eis a cache mount anyway, it won't persist in the image
+	_conf_update_success: core.#WriteFile & {
+		input:    dagger.#Scratch
+		contents: "APT::Update::Post-Invoke-Success { \"mkdir -p /var/cache/apt/dagger; touch \(updated_path)\"; };"
+		path:     "/apt-update-invoke-success"
+	}
+
+	// Use this to mount over some well-meaning default apt configs that get injected into the base contianer image.
+	// The configs do things like clean up the apt cache after install in order to keep image sizes smaller.
+	// For the caching strategy here, it is desirable to keep the apt cache in tact as it is getting written to a cache mount instead of the container fs.
+	_empty_file: core.#WriteFile & {
+		input:    dagger.#Scratch
+		path:     "/empty"
+		contents: ""
+	}
+
+	mounts: {
+		"no-apt-clean": {
+			type:     "fs"
+			ro:       true
+			contents: _empty_file.output
+			source:   _empty_file.path
+			dest:     "/etc/apt/apt.conf.d/docker-clean"
+		}
+		"no-apt-gzip": {
+			type:     "fs"
+			ro:       true
+			contents: _empty_file.output
+			source:   _empty_file.path
+			dest:     "/etc/apt/apt.conf.d/docker-gzip-indexes"
+		}
+		"keep-cache": {
+			type:     "fs"
+			ro:       true
+			contents: _conf_keep_cache.output
+			source:   _conf_keep_cache.path
+			dest:     "/etc/apt/apt.conf.d/dagger-io-keep-cache"
+		}
+		"update-success": {
+			type:     "fs"
+			ro:       true
+			contents: _conf_update_success.output
+			source:   _conf_update_success.path
+			dest:     "/etc/apt/apt.conf.d/dagger-io-update-success-trigger"
+		}
+		"var-cache-apt": {
+			contents: core.#CacheDir & {
+				id: "\(cache_prefix)-var-cache-apt"
+			}
+			dest: "/var/cache/apt"
+		}
+		"var-lib-apt": {
+			contents: core.#CacheDir & {
+				id: "\(cache_prefix)-var-lib-apt"
+			}
+			dest: "/var/lib/apt"
+		}
+	}
+}
+
+#Install: {
+	input:        docker.#Image
+	cache_prefix: string | *""
+	packages: [pkgName=string]: {}
+
+	_apt: _#aptMounts & {
+		"cache_prefix": cache_prefix
+	}
+
+	docker.#Build & {
+		steps: [
+			{output: input},
+			for pkgName, pkg in packages {
+				docker.#Run & {
+					mounts: _apt.mounts
+					env: {
+						DEBIAN_FRONTEND:     "noninteractive"
+						PKG_NAME:            pkgName
+						DAGGER_UPDATED_PATH: _apt.updated_path
+					}
+					command: {
+						name: "/bin/sh"
+						args: ["-c", """
+                            set -e
+                            if [ -z "$(find ${DAGGER_UPDATED_PATH} -mmin -60)" ]; then 
```

Comment:
1. This is neat
2. Do you think it would make sense to make the `60` configurable? In case someone wants to force a re-run of apt update? Not necessary if it's hard for whatever reason, just a thought in case it's easy.

Reply:
Added.